### ISSUE TITLE
Data loss when creating documents with null value two nested levels deep and strict false

### ISSUE THREAD

**Issue by jakesjews** [Automattic/mongoose#6187]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
When creating a document with nested schemas two levels deep if the lowest nested document is null then the parent doesn't have a value even if other keys are set. The issue seems to only occur when strict is false on the nested schema. The issue only happens in versions greater than 5.0.4.

**If the current behavior is a bug, please provide the steps to reproduce.**
```
  describe('nested fields with a schema', function() {
    it('saves', function(done) {
      var NestedSchema = new Schema({
        parent: new Schema({
          name: String,
          child: {
            name: String
          }
        }, { strict: false })
      });
      var NestedModel = db.model('Nested', NestedSchema);
      var n = new NestedModel({
        parent: {
          name: 'foo',
          child: null // does not fail if undefined
        }
      });

      assert.equal(n.parent.name, 'foo');
      done();
    });
  });
```

**What is the expected behavior?**
The parent nested object has a value.

**Please mention your node.js, mongoose and MongoDB version.**
node: 9.6.1
mongoose: 5.0.7
mongodb: 3.6.3

**Comment by lineus** [comment]
I was able to reproduce this:
```
#!/usr/bin/env node
'use strict'

const mongoose = require('./lib/test_db')
const Schema = mongoose.Schema
const assert = require('assert')

const nestedSchema = new Schema({
  parent: new Schema({
    name: String,
    child: {
      name: String
    }
  }, { strict: false })
})

const Nested = mongoose.model('nested', nestedSchema, 'nested')

const nest = new Nested({
  parent: new Nested({
    name: 'foo',
    child: null
  })
})

nest.save((err, res) => {
  if (err) { return console.error(err) }
  assert.equal(res.parent.name, 'foo')
  require('get-mongoose-versions')
    .get('mongodb://localhost/test')
    .then(versions => {
      mongoose.connection.close()
      console.log(versions.mongooseVersion, 'no assertion errors')
    })
})
```
the output:
```
InspiredMacPro:mongoose5 lineus$ npm install mongoose@5.0.2 >/dev/null 2>&1 && ./6187.js
/Users/lineus/dev/Help/mongoose5/node_modules/mongodb/lib/utils.js:132
      throw err;
      ^

AssertionError [ERR_ASSERTION]: undefined == 'foo'
    at nest.save (/Users/lineus/dev/Help/mongoose5/6187.js:28:10)
    at /Users/lineus/dev/Help/mongoose5/node_modules/mongoose/lib/model.js:3928:16
    at model.$__save.error (/Users/lineus/dev/Help/mongoose5/node_modules/mongoose/lib/model.js:343:7)
    at /Users/lineus/dev/Help/mongoose5/node_modules/kareem/index.js:297:21
    at next (/Users/lineus/dev/Help/mongoose5/node_modules/kareem/index.js:209:27)
    at Kareem.execPost (/Users/lineus/dev/Help/mongoose5/node_modules/kareem/index.js:217:3)
    at _cb (/Users/lineus/dev/Help/mongoose5/node_modules/kareem/index.js:289:15)
    at $__handleSave (/Users/lineus/dev/Help/mongoose5/node_modules/mongoose/lib/model.js:281:5)
    at /Users/lineus/dev/Help/mongoose5/node_modules/mongoose/lib/model.js:168:7
    at session.endSession (/Users/lineus/dev/Help/mongoose5/node_modules/mongodb/lib/utils.js:400:34)
    at ClientSession.endSession (/Users/lineus/dev/Help/mongoose5/node_modules/mongodb-core/lib/sessions.js:69:41)
    at args.push (/Users/lineus/dev/Help/mongoose5/node_modules/mongodb/lib/utils.js:397:17)
    at /Users/lineus/dev/Help/mongoose5/node_modules/mongodb/lib/collection.js:652:5
    at args.push (/Users/lineus/dev/Help/mongoose5/node_modules/mongodb/lib/utils.js:404:72)
    at handleCallback (/Users/lineus/dev/Help/mongoose5/node_modules/mongodb/lib/utils.js:128:55)
    at /Users/lineus/dev/Help/mongoose5/node_modules/mongodb/lib/bulk/unordered.js:553:16
```
I get the same assertion error on all versions of mongoose@5 and @4.13.11
with `strict: false` on the inner schema. setting `strict: false` on the outer schema instead works as expected.

**Comment by vkarpov15** [comment]
Fixed in master and will be in 5.0.10. Issue was a validation error, if you tried to `n.save()` you would get a validation error that tracks down to an error in mongoose's internal `applyDefaults()` logic

**Comment by jakesjews** [comment]
@vkarpov15 thanks a ton for looking into this!
